### PR TITLE
[TTAHUB-3060] Self Service Automation - Database Backup - Phase 2: Publish production url in slack

### PR DESCRIPTION
**Pull request by GarrettEHill**
## Description of change
When production is backed up a message is posted to acf-head-start-eng
Refactored deployment code to post successful deployments of Staging, Sandbox, Dev to acf-head-start-github 

## How to test
look for messages in slack

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3060

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: bin/latest_backup.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
```
Comment (GarrettEHill) [inline bin/latest_backup.sh:1]:
```suggestion
#!/bin/bash
```

File: bin/latest_backup.sh [hunk 2]
```diff
@@ -51,11 +51,13 @@ fetch_service_key() {
     local cf_s3_service_name=$1
     local key_name=$2
     local full_output=$(cf service-key "${cf_s3_service_name}" "${key_name}" 2>&1)
+    echo "Service key full output: $full_output"
```
Comment (GarrettEHill) [inline bin/latest_backup.sh:54]:
```suggestion
```

File: bin/latest_backup.sh [hunk 3]
```diff
@@ -51,11 +51,13 @@ fetch_service_key() {
...
     local credentials_json=$(echo "${full_output}" | awk '/\{/,0')
     if [ -z "${credentials_json}" ]; then
         echo "No JSON data found." >&2
         exit 5
     fi
+    echo "Credentials JSON: $credentials_json"
```
Comment (GarrettEHill) [inline bin/latest_backup.sh:60]:
```suggestion
```

File: bin/latest_backup.sh [hunk 4]
```diff
@@ -145,6 +147,12 @@ fetch_latest_backup_info_and_cleanup() {
     local aws_default_region=$(echo "${credentials_json}" | jq -r '.credentials.region')
     local bucket_name=$(echo "${credentials_json}" | jq -r '.credentials.bucket')
 
+    # Debugging output
+    echo "AWS_ACCESS_KEY_ID: $aws_access_key_id"
+    echo "AWS_SECRET_ACCESS_KEY: $aws_secret_access_key"
+    echo "AWS_DEFAULT_REGION: $aws_default_region"
+    echo "BUCKET_NAME: $bucket_name"
+
```
Comment (GarrettEHill) [inline bin/latest_backup.sh:155]:
```suggestion
```

File: automation/ci/scripts/cf_lambda.sh [hunk 5]
```diff
@@ -467,11 +470,56 @@ function delete_app {
...
+                    log "ERROR" "File $file_name verification failed."
+                    if [ -f "$output_directory/$file_name" ]; then
+                      rm -rf "$output_directory/$file_name"
+                    fi
+                fi
+            fi
+        fi
+    done <<< "$logs"
+}
+
```
Comment (GarrettEHill) [inline automation/ci/scripts/cf_lambda.sh:517]:
```suggestion
```

File: automation/ci/scripts/cf_lambda.sh [hunk 6]
```diff
@@ -467,11 +470,56 @@ function delete_app {
...
 # -----------------------------------------------------------------------------
 
 main() {
   # Check dependencies first
-  check_dependencies cf awk date grep jq sleep uuidgen
+  check_dependencies cf awk date grep jq sleep uuidgen base64 sha256sum md5sum
```
Comment (GarrettEHill) [inline automation/ci/scripts/cf_lambda.sh:522]:
```suggestion
  check_dependencies cf awk date grep jq sleep uuidgen
```

File: .circleci/config.yml [hunk 7]
```diff
@@ -1001,16 +1141,18 @@ jobs:
...
+                environment_name: "production"
+
     resource_class: large
   backup_upload_production:
     docker:
       - image: cimg/base:2024.05
     steps:
     - sparse_checkout:
-          directory: 'automation'
+          directories: 'automation,bin'
```
Comment (GarrettEHill) [inline .circleci/config.yml:1155]:
```suggestion
          directories: 'automation'
```

File: .circleci/config.yml [hunk 8]
```diff
@@ -308,15 +411,19 @@ commands:
...
             echo "deb https://packages.cloudfoundry.org/debian stable main" | sudo tee /etc/apt/sources.list.d/cloudfoundry-cli.list
             sudo apt-get update
-            sudo apt-get install cf8-cli
+            sudo apt-get install -y cf8-cli
             # Install plugin needed for connect-to-service
             cf install-plugin -f https://github.com/cloud-gov/cf-service-connect/releases/download/v1.1.3/cf-service-connect_linux_amd64
 
+            # Install AWS CLI
+            sudo apt-get install -y awscli
+
```
Comment (GarrettEHill) [inline .circleci/config.yml:426]:
```suggestion
```

**Comment by GarrettEHill** [comment]
Note: there is no way to get valid URLs to slack, all methods proved fruitless. URLs generated were invalidated by the time the CI job finished.

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**